src/DDTrace/Integrations/Predis/PredisIntegration.php

```diff
@@ -158,8 +158,11 @@ public static function storeConnectionParams($predis, $args)
 
         if (isset($args[1])) {
             $options = $args[1];
-            if (isset($options['parameters']) && isset($options['parameters']['database'])) {
-                $tags['out.redis_db'] = $options['parameters']['database'];
+
+            $parameters = $options->__get('parameters');
```

Comment:
Thanks for noticing this. Actually it seems to me that options can be provided both as an array and as an object instance: https://github.com/nrk/predis/blob/111d100ee389d624036b46b35ed0c9ac59c71313/src/Client.php#L71

So what do you think about doing something like:

```
if (is_array($options)) {
    $parameters = isset($options['parameters']) ? $options['parameters'] : [];
} elseif ($options instanceof OptionsInterface) {
    $parameters = $options->__get('parameters') ?: [];
}
```

If you like the idea, would you please add a test case for when we have options as object?
https://github.com/DataDog/dd-trace-php/blob/455e89ac54528eb7129e193ddc096bbf67171074/tests/Integration/Integrations/Predis/PredisTest.php#L42

Reply:
Hi Luca, it seems a good solution, I like it! I'll add the test case asap, thank you.